src/Service/MentoringManager.php

```diff
@@ -66,14 +73,33 @@ public function startMentoring(Mentoring $mentoring): void
     }
 
     /**
-     * to use whenever you need to unaccept a mentoring
+     * to use whenever you need to stop a mentoring
+     * set ending date to today's date, is accepted to false and remove all messages.
      */
     public function stopMentoring(Mentoring $mentoring): void
     {
         $mentoringToUpdate = $this->mentoringRepository->find($mentoring);
         if ($mentoringToUpdate !== null) {
             $mentoringToUpdate->setIsAccepted(false);
+            $mentoringToUpdate->setEndingDtae(new DateTime());
             $this->entityManager->flush();
+            $this->removeMessages($mentoringToUpdate);
+        }
+    }
+
+    /**
+     * remove all messages for a specific mentoring
+     */
+    public function removeMessages(Mentoring $mentoring): void
+    {
```

Comment:
Why don't you use the removeMessage method that already exists in Mentoring Entity ?